matrix-transpose/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+#![cfg_attr(
+    feature = "simd-transpose",
+    feature(slice_split_at_unchecked),
+    feature(portable_simd),
+    feature(stmt_expr_attributes),
+    feature(slice_as_chunks)
+)]
+
+#[cfg(not(feature = "simd-transpose"))]
+mod scalar;
+#[cfg(feature = "simd-transpose")]
+mod simd;
+
+#[cfg(feature = "simd-transpose")]
+pub use simd::transpose_unchecked;
+
+#[cfg(not(feature = "simd-transpose"))]
+pub use scalar::transpose_unchecked;
+
+use thiserror::Error;
+
+/// This function transposes a matrix on the bit-level.
+///
+/// This implementation requires that the number of rows is a power of 2
+/// and that the number of columns is a multiple of 8
+pub fn transpose_bits(matrix: &mut [u8], rows: usize) -> Result<(), TransposeError> {
+    // Check that slice is rectangular
```

Comment:
I feel like not everyone is familiar with using & like that. Let's add a comment:
// Check that slice is rectangular, i.e. the number of cells is  a multiple of the number of rows.